pkg/csi/service/vanilla/controller.go

```diff
@@ -760,6 +757,25 @@ func (c *controller) ControllerExpandVolume(ctx context.Context, req *csi.Contro
 	log := logger.GetLogger(ctx)
 	log.Infof("ControllerExpandVolume: called with args %+v", *req)
 
+	var volumePath string
+	if strings.Contains(req.VolumeId, ".vmdk") {
+		volumePath = req.VolumeId
+	}
+	if containerOrchestratorUtility.IsFSSEnabled(ctx, common.CSIMigration) {
+		// Migration feature switch is enabled
+		if volumePath != "" {
```

Comment:
You can move this `if volumePath != ""` check outside of `if containerOrchestratorUtility.IsFSSEnabled(ctx, common.CSIMigration){}`.
This will avoid code duplication

Reply:
done.

```
        if volumePath != "" {
		if containerOrchestratorUtility.IsFSSEnabled(ctx, common.CSIMigration) {
			msg := fmt.Sprintf("Cannot expand migrated vSphere volume. :%q", volumePath)
			log.Error(msg)
			return nil, status.Errorf(codes.Unimplemented, msg)
		} else {
			msg := fmt.Sprintf("volume-migration feature switch is disabled. Cannot use volume with vmdk path :%q", volumePath)
			log.Error(msg)
			return nil, status.Errorf(codes.Internal, msg)
		}
	}
```